### PR TITLE
DOC: Clean dubious docstring

### LINKED ISSUE #50344: CLEAN dubious docstring
This looks odd - the docstring is broken in two, and there's an assignment in the middle?

https://github.com/pandas-dev/pandas/blob/22491dc041315d40e525b722a92202f656639957/pandas/_libs/tslibs/offsets.pyx#L369-L389

It looks like something might not have gone quite right during a merge conflict, and was kept in?

If anyone felt like having a look at `git blame` / `git log` to figure this out, that'd be helpful

No need to comment "can I work on this?" - if you know your way around `git` and figure this out, then go ahead and work on it - thanks 🙏

### PR DESCRIPTION
**Pull request by luke396**
Fixes #50344

- [x] closes #50344

### CODE REVIEW

**Comment by datapythonista** [comment]
Thanks @luke396 this looks nice. I think in general we don't document `__init__` but the class itself. Do you mind moving what's in the `__init__` docstring to the class docstring? I think that will be even better and more consistent. Thanks!

**Comment by luke396** [comment]
> Thanks @luke396 this looks nice. I think in general we don't document `__init__` but the class itself. Do you mind moving what's in the `__init__` docstring to the class docstring? I think that will be even better and more consistent. Thanks!

I change it, what about this time? 😃

File: pandas/_libs/tslibs/offsets.pyx [hunk 1]
```diff
@@ -366,6 +366,22 @@ class ApplyTypeError(TypeError):
 cdef class BaseOffset:
     """
     Base class for DateOffset methods that are not overridden by subclasses.
+
+    Parameters:
+    -----------
```
Comment (datapythonista) [inline pandas/_libs/tslibs/offsets.pyx:371]:
```suggestion
    Parameters
    ----------
```

File: pandas/_libs/tslibs/offsets.pyx [hunk 2]
```diff
@@ -366,6 +366,22 @@ class ApplyTypeError(TypeError):
...
+    n : int
+        Number of multiples of the frequency.
+
+    normalize : boolean
```
Comment (datapythonista) [inline pandas/_libs/tslibs/offsets.pyx:375]:
```suggestion
    normalize : bool
```

File: pandas/_libs/tslibs/offsets.pyx [hunk 3]
```diff
@@ -366,6 +366,22 @@ class ApplyTypeError(TypeError):
...
+        Whether the frequency can align with midnight.
+
+    Examples
+    --------
+    >>> pd.offsets.Hour(5).n
+    5
+
```
Comment (datapythonista) [inline pandas/_libs/tslibs/offsets.pyx:382]:
```suggestion
```
If I remember correctly, blank lines in the examples cause two separate blocks to be rendered, which won't probably look very nice in this case.

**Review by datapythonista — COMMENTED** [review]
Looks great. Just added couple of small things, you can commit them from the github UI

**Comment by luke396** [comment]
> Looks great. Just added couple of small things, you can commit them from the github UI

Thanks for the review!

**Review by datapythonista — APPROVED** [review]
lgtm, thanks for fixing this @luke396

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx @luke396

**Pull request merged**